### PR TITLE
Hotfix scrm 645 use personal email

### PR DESCRIPTION
**Pull request by gymad**
## Description

Emails: Sending emails uses personal instead of system outbound by default

## How To Test This

1. Setup a personal inbound/outbound email account.
2. Set a non-working outgoing mail configuring in admin panel "email settings".
3. Attempt to send an email.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [x] My code follows the code style of this project found [here](https://suitecrm.com/wiki/index.php/Coding_Standards).
- [x] My change requires a change to the documentation.
- [x] I have read the [**How to Contribute**](https://suitecrm.com/wiki/index.php/Contributing_to_SuiteCRM) guidelines.

### CODE REVIEW

File: include/OutboundEmail/OutboundEmail.php [hunk 1]
```diff
@@ -249,30 +249,42 @@ function getUserMailers($user) {
...
 				if(isset($opts['outbound_email'])) {
 					$mailer = "AND id = '{$opts['outbound_email']}'";
 				}
 			}
 		}
 
-		$q = "SELECT id FROM outbound_email WHERE user_id = '{$user->id}' {$mailer}";
-		$r = $this->db->query($q);
-		$a = $this->db->fetchByAssoc($r);
+		$query = "SELECT * FROM outbound_email WHERE user_id = '{$user->id}' {$mailer} AND deleted = 0;";
```
Comment (sergio91pt) [inline include/OutboundEmail/OutboundEmail.php]: Why change `SELECT id` to `SELECT *`?

Comment (daniel-samson) [inline include/OutboundEmail/OutboundEmail.php]: Can you answer @sergio91pt question?

Comment (gymad) [inline include/OutboundEmail/OutboundEmail.php]: Oh, right question :) fortunately it doesn't change any behavior. I revert it back

File: modules/Emails/Email.php [hunk 2]
```diff
@@ -2695,8 +2695,16 @@ public function send()
             }
         }
 
+        $mailBefore = clone $mail;
+
         $mail = $this->setMailer($mail);
 
+        // use personal email by default
+
+        if($mailBefore->oe->type === 'user' && $mail->oe->type === 'system') {
```
Comment (samus-aran) [inline modules/Emails/Email.php]: Do we need to consider the system_override?

Why mailBefore?  Why we checking the mail account before and after for?

**Comment by edwinlee2000** [comment]
I applied hotfix to 7.9.7 and still doesn't work.
Configured both system and personal outgoing email, tested ok.
But got error when trying to compose, then send.
UI showed: "Error Sending Email. Please contact your administrator for assistance."
Log showed: [SECURITY] User is not allowed to send email
Reason was:
User is not allowed to use outbound email.
[SECURITY] User XXX attempted to send an email using incorrect email account settings in which they do not have access to.
It seems still haven't resolved the "Outgoing personal email" issue !!
Anyone can help !

**Comment by Dillon-Brown** [comment]
Superceded by #6017

**Pull request closed without merging**